dynawo/sources/Common/DYNTrace.cpp

```diff
@@ -255,12 +257,12 @@ void Trace::resetPersistantCustomAppender_(const std::string& tag, SeverityLevel
   set.insert("Thread", attrs::make_attribute_value(currentId));
 
   TraceSinks& traceSink = sinks_.at(currentId);
-  for (vector< boost::shared_ptr<FileSink> >::const_iterator itSinks = traceSink.persistantSinks.begin();
-    itSinks != traceSink.persistantSinks.end(); ++itSinks) {
-    if ((*itSinks)->will_consume(set))
-      logging::core::get()->remove_sink(*itSinks);
+  for (auto itSinks = traceSink.persistantSinks.cbegin(); itSinks != traceSink.persistantSinks.cend(); ++itSinks) {
+    if (itSinks->second->will_consume(set))
+      logging::core::get()->remove_sink(itSinks->second);
   }
-  traceSink.persistantSinks.clear();
+  TraceSinks::TagAndSeverityLevel tagAndSeverityLevel = std::make_pair(tag, slv);
```

Comment:
What happens if we dump in 2 different files the same tag end severity level?

Reply:
With the current code, the **sinks** attribute of the **TraceSinks** struct isn't a multimap. So if log files have the same tag and severity level, they will differ eventually.

Reply:
this scenario was supported before this development, no reason to forbid it after as it should be a transparent development. I still thinks that using the will_consume method to know which sinks need to be removed is more robust

Reply:
Currently, the sinks are removed by using **logging::core::get()->remove_sink(sink)**

Reply:
then what is the point of using a map?

Reply:
Moreover if we have 2 identical appenders with only the file name that is different, one of them will not be removed

Reply:
> then what is the point of using a map?

When we want to delete a specific sink, we also need to delete a specific element in the **TraceSinks** struct. The map is the easiest way to do this.

> Moreover if we have 2 identical appenders with only the file name that is different, one of them will not be removed

The std::multimap<Key,T>::erase() method erases every elements which have the same key.

Reply:
ok got it